File: src/main/java/com/example/installcertatruntime/user/UserController.java
package com.example.installcertatruntime.user;

import org.springframework.web.bind.annotation.*;

import java.util.List;

import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;

@RestController
@RequestMapping(value = "api/v1/users")
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("")
    public List<User> all() {
        return userService.all();
    }

    @PostMapping(value = "", consumes = APPLICATION_JSON_UTF8_VALUE)
    public void create(@RequestBody(required = false) User user) {
        userService.create(user);
    }

}

File: src/main/kotlin/com/example/installcertatruntime/user/UserController.kt
package com.example.installcertatruntime.user

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["api/v1/users"])
class UserController(private val userService: UserService) {

    @GetMapping("")
    fun all(): List<User> {
        return userService.all()
    }

    @PostMapping(value = [""])
    fun create(@RequestBody(required = false) user: User) {
        userService.create(user)
    }

}
